bin/pqcheck

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bin/ruby
+
+DEFAULT_PQ_DIR = "data/queue/main"
+
+argv0 = ARGV[0].to_s.strip
+if argv0 == "-h" || argv0 == "--help"
+  puts("usage: pqcheck [PQ dir path]\n  default [PQ dir path] is #{DEFAULT_PQ_DIR}")
+  exit(0)
+end
+
+dir = argv0.empty? ? DEFAULT_PQ_DIR : argv0
+if (!Dir.exist?(dir))
+  puts("error: invalid PQ dir path: #{dir}")
+  exit(1)
+end
+
+puts("checking queue dir: #{dir}")
+
+Dir.glob("#{dir}/checkpoint.*").sort_by { |x| x[/[0-9]+$/].to_i }.each do |checkpoint|
+  data = File.read(checkpoint)
+  version, page, firstUnackedPage, firstUnackedSeq, minSeq, elementCount, crc32 = (data.size == 33) ? data.unpack("nNNQ>Q>NN") : [0,0,0,0,0,0,0]
```

Comment:
if the checkpoint is not 33 bytes, we proceed as if it were encoded with all zeroes; should we instead emit a warning, or are there cases where it is valid for the checkpoint to be a different size?

Reply:
true, it's not implicitly written it is an invalid checkpoint but there is not such all-zeros checkpoints except for newly created & empty queue...